src/redux/todosSlice.ts

```diff
@@ -0,0 +1,73 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+export interface Todo {
+  id: string;
+  text: string;
+  completed: boolean;
+}
+
+interface TodosState {
+  todos: Todo[];
+}
+
+const saveState = (state: Todo[]) => {
+  try {
+    const serializedState = JSON.stringify(state);
+    localStorage.setItem('todos', serializedState);
+  } catch (error) {
+    console.error('Error saving state:', error);
+  }
+};
+
+const loadState = (): TodosState => {
+  const emptyState = {
+    todos: [],
+  };
+  try {
+    const serializedState = localStorage.getItem('todos');
+    if (serializedState === null) {
+      return emptyState;
+    }
+    return {
+      todos: JSON.parse(serializedState),
+    };
+  } catch (error) {
+    console.error('Error loading state:', error);
+    return emptyState;
+  }
+};
+
+const initialState: TodosState = loadState();
+
+const todosSlice = createSlice({
+  name: 'todosList',
+  initialState,
+  reducers: {
+    addTodo: (state, action: PayloadAction<string>) => {
+      const newTodo: Todo = {
+        id: Date.now().toString(),
```

Comment:
There is a good `nanoid` library for this

Reply:
Nanoid has been added